app/controllers/research/user_experiments_controller.rb

```diff
@@ -0,0 +1,20 @@
+module Research
+  class UserExperimentsController < Research::BaseController
+    def create
+      experiment = Experiment.find(params[:experiment_id])
+      begin
```

Comment:
Why do we have this wrapped?

Reply:
To stop race conditions when people double click (See the test for it).